platforms/org.eclipse.mita.platform.xdk110/src/org/eclipse/mita/platform/xdk110/connectivity/MqttGenerator.xtend

```diff
@@ -233,19 +233,19 @@ class MqttGenerator extends AbstractSystemResourceGenerator {
 		return retcode;
 		''');
 		if(auth instanceof SumTypeRepr) {
-			if(auth.name == "Login") {
+			if(auth.isLogin()) {
 				val username = auth.properties.get("username")?.code;
 				val password = auth.properties.get("password")?.code;
 				result.setPreamble('''
 				const char usernameBuf[] = «username»;
 				StringDescr_T username = {
```

Comment:
I know that this is my code, but while you're at it: could you use `StringDescr_wrap` here? See for example line ~370.